src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -636,61 +623,67 @@ private static BlittableJsonReaderObject RevertSnapshotFlags(DocumentsOperationC
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice lowerId, CollectionName collectionName,
-            RevisionsCollectionConfiguration configuration, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks)
+        private void DeleteOldRevisionsForPutOrDelete(DocumentsOperationContext context, Table table, Slice lowerId, CollectionName collectionName,
+            RevisionsCollectionConfiguration configuration, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks,
+            bool documentDeleted = false)
         {
             using (GetKeyPrefix(context, lowerId, out Slice prefixSlice))
             {
                 // We delete the old revisions after we put the current one,
                 // because in case that MinimumRevisionsToKeep is 3 or lower we may get a revision document from replication
                 // which is old. But because we put it first, we make sure to clean this document, because of the order to the revisions.
                 var revisionsCount = IncrementCountOfRevisions(context, prefixSlice, 1);
-                DeleteOldRevisions(context, table, prefixSlice, collectionName, configuration, revisionsCount, nonPersistentFlags, changeVector, lastModifiedTicks);
+                DeleteOldRevisions(context, table, prefixSlice, collectionName, configuration, nonPersistentFlags, changeVector, lastModifiedTicks, documentDeleted: documentDeleted);
             }
         }
 
-        private bool DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, CollectionName collectionName,
-            RevisionsCollectionConfiguration configuration, long revisionsCount, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks)
+        public class DeleteOldRevisionsResult
+        {
+            public bool HasMore;
+            public long Remaining;
+        }
+
+        private DeleteOldRevisionsResult DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, CollectionName collectionName,
+            RevisionsCollectionConfiguration configuration, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks,
+            bool documentDeleted, bool skipForceCreated = false)
         {
-            var moreRevisionToDelete = false;
+            var revisionsCount = GetRevisionsCount(context, prefixSlice);
```

Comment:
should this be the initial value of the `result.Remaining` ?

Reply:
Done